python/composio/core/models/tools.py

```diff
@@ -85,10 +85,12 @@ def _filter_custom_tools(
         """Filter out custom tools from the list of tools."""
         _tools = []
         _custom_tools = []
+        custom_tools_registry = self._custom_tools.custom_tools_registry
         for tool in tools:
-            try:
-                _custom_tools.append(self._custom_tools[tool].info)
-            except KeyError:
+            custom_tool = custom_tools_registry.get(tool)
```

Comment:
Thanks for the improvement @Saga4

Can we instead update the get function of `CustomTools` itself to do a `custom_tools_registry.get(tool)` instead ?